### ISSUE TITLE
How to install Smaragd in KDE Neon

### ISSUE THREAD

**Issue by cinnamoner** [cinnamoner/Tutorials#1]
*Level of proficiency: noob*

**Comment by cinnamoner** [comment]
Timeframe: soon

**Comment by cinnamoner** [comment]
@cfeck @MichaelTunnell @BigDaddyonTech 
Ready for some rock 'n roll? Tutorial unfolding in real time. I have no idea how it will go. I think those are the required packages. Have I missed any?

![screenshot_20170307_175422](https://cloud.githubusercontent.com/assets/15638605/23665241/9aea8ada-0360-11e7-8265-6983d93353a7.png)

**Comment by cinnamoner** [comment]
![f66d9382fbbf2619240774c47700515da863](https://cloud.githubusercontent.com/assets/15638605/23665784/1db37976-0362-11e7-9dd2-59dd7af0d599.png)


Oh, https://store.kde.org/p/1167274/

[smaragd-0.1.0.tar.gz](https://github.com/cinnamoner/Tutorials/files/825004/smaragd-0.1.0.tar.gz)
SHA256: 464ae1b07b4d15614a332b447e9227aaff4ab261c82f215dd1448f8a76c12e78

**Comment by cinnamoner** [comment]
@cfeck Confirm it, please. I'm not going forth without your approval.

**Comment by BigDaddyonTech** [comment]
I would help you out but that is not my area of expertise

**Comment by cinnamoner** [comment]
> I would help you out but that is not my area of expertise

@BigDaddyonTech You're welcome nonetheless. Maybe we'll shoot a video, should Christoph agree to it.

**Comment by cfeck** [comment]
We already investigated that you also need to install "extra-cmake-modules".

**Comment by cfeck** [comment]
I really would prefer if you just fetch the source from git as described at https://kdepepo.wordpress.com/2016/12/23/smaragd-for-plasma-5/

This ensures users following the tutorial always have the newest version.

**Comment by cinnamoner** [comment]
> We already investigated that you also need to install "extra-cmake-modules".

![screenshot_20170307_193408](https://cloud.githubusercontent.com/assets/15638605/23669544/404b230c-036d-11e7-9567-fdacbf4f6093.png)

@cfeck Like that, you mean?

**Comment by cinnamoner** [comment]
I basically want all the correct packages after *sudo-apt get install*, if possible. Excuse my ignorance.

**Comment by cinnamoner** [comment]
@cfeck I have read what your link leads to a number of times but I still don't understand what I have to do concretely. Show it to me in the terminal, please.

**Comment by cinnamoner** [comment]
@Zren What on earth do I type into the terminal so I can build the thing? Sorry for the disturbance, if any.

**Comment by cinnamoner** [comment]
> CMake Error at CMakeLists.txt:5 (find_package):
Could not find a package configuration file provided by "ECM" (requested
version 1.0.0) with any of the following names:

> ECMConfig.cmake
ecm-config.cmake

> Add the installation prefix of "ECM" to CMAKE_PREFIX_PATH or set "ECM_DIR"
to a directory containing one of the above files. If "ECM" provides a
separate development package or SDK, be sure it has been installed.

> -- Configuring incomplete, errors occurred!

**Comment by cfeck** [comment]
If you still get the error that ECM is not found, then you have not installed extra-cmake-modules.

**Comment by cinnamoner** [comment]
@cfeck What should I understand out of this screenshot?
![screenshot_20170307_221701](https://cloud.githubusercontent.com/assets/15638605/23676177/8060176a-0384-11e7-988b-a6696c44ba8c.png)

**Comment by cfeck** [comment]
You also need to install "gettext-dev". If you get more errors, use a text paste instead of a screen shot.

Additionally, do not use kde4-config, that is invalid in Plasma5. Just use /usr as the prefix.

**Comment by cinnamoner** [comment]
> You also need to install "gettext-dev".

Cool. "Sudo apt-get install gettext-dev"

> Additionally, do not use kde4-config, that is invalid in Plasma5. Just use /usr as the prefix.

Ha?

**Comment by cinnamoner** [comment]
@cfeck ?

> Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package gettext-dev

**Comment by cinnamoner** [comment]
I'm getting scared :)

**Comment by cfeck** [comment]
Oh, the package is just called "gettext"

**Comment by cinnamoner** [comment]
@cfeck 

> Readingpackage lists... Done
Building dependency tree       
Reading state information... Done
Starting pkgProblemResolver with broken count: 0
Starting 2 pkgProblemResolver with broken count: 0
Done
The following packages were automatically installed and are no longer required:
  calligra-data calligrawords-data kdiagram-data krita-data libcauchy0.0v5 libetonyek-0.1-1 libkchart2 liblangtag-common liblangtag1 libm2mml0.0v5 libodfgen-0.1-1 libopencolorio1v5 libphonon4qt5experimental4 libspnav0 libtinyxml2.6.2v5 libwps-0.4-4 libyaml-cpp0.3v5
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  libasprintf-dev libgettextpo-dev libgettextpo0 libunistring0
Suggested packages:
  gettext-doc autopoint
The following NEW packages will be installed:
  gettext libasprintf-dev libgettextpo-dev libgettextpo0 libunistring0
0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
Need to get 1.626 kB of archives.
After this operation, 6.707 kB of additional disk space will be used.
Do you want to continue? [Y/n]

**Comment by cinnamoner** [comment]
> The following packages were automatically installed and are no longer required:
calligra-data calligrawords-data kdiagram-data krita-data libcauchy0.0v5 libetonyek-0.1-1 libkchart2 liblangtag-common liblangtag1 libm2mml0.0v5 libodfgen-0.1-1 libopencolorio1v5 libphonon4qt5experimental4 libspnav0 libtinyxml2.6.2v5 libwps-0.4-4 libyaml-cpp0.3v5
Use 'sudo apt autoremove' to remove them.

Done.

**Comment by cinnamoner** [comment]
@cfeck 

> richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0$ cd smaragd
bash: cd: smaragd: No such file or directory
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0$ mkdir build
mkdir: cannot create directory ‘build’: File exists
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0$ cd build
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0/build$ cmake -DCMAKE_INSTALL_PREFIX=`kde4-config --prefix` ..
-- Found Gettext: /usr/bin/msgmerge (found version "0.19.7") 
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Found KF5I18n: /usr/lib/x86_64-linux-gnu/cmake/KF5I18n/KF5I18nConfig.cmake (found version "5.31.0") 
-- Found KF5WidgetsAddons: /usr/lib/x86_64-linux-gnu/cmake/KF5WidgetsAddons/KF5WidgetsAddonsConfig.cmake (found version "5.31.0") 
-- Found KF5: success (found version "5.31.0") found components:  Config CoreAddons I18n WidgetsAddons 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found GLIB2: /usr/lib/x86_64-linux-gnu/libglib-2.0.so  
-- Checking for module 'cairo'
--   Found cairo, version 1.14.6
-- Found Cairo: /usr/lib/x86_64-linux-gnu/libcairo.so  
-- Installing in the same prefix as Qt, adopting their path scheme.
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /home/richard/Downloads/smaragd-0.1.0/build
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0/build$ make
Scanning dependencies of target kwin_smaragd_automoc
[  6%] Automatic moc for target kwin_smaragd
Generating kwin_smaragd.moc
Generating moc_kwin_smaragd.cpp
[  6%] Built target kwin_smaragd_automoc
[ 13%] Generating ui_kwin_smaragd_dialog.h
Scanning dependencies of target kwin_smaragd
[ 20%] Building CXX object src/CMakeFiles/kwin_smaragd.dir/kwin_smaragd.cpp.o
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp: In member function ‘virtual void Smaragd::Decoration::paint(QPainter*, const QRect&)’:
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:732:16: warning: unused variable ‘horizontalBorders’ [-Wunused-variable]
     const bool horizontalBorders = !client().data()->isMaximizedHorizontally();
                ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:733:16: warning: unused variable ‘verticalBorders’ [-Wunused-variable]
     const bool verticalBorders = !client().data()->isMaximizedVertically();
                ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:735:11: warning: variable ‘decoSize’ set but not used [-Wunused-but-set-variable]
     QSize decoSize = size();                                                                                                                                                                                                                                                  
           ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:748:9: warning: unused variable ‘textHaloXOffset’ [-Wunused-variable]
     int textHaloXOffset = 1;                                                                                                                                                                                                                                                  
         ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:749:9: warning: unused variable ‘textHaloYOffset’ [-Wunused-variable]
     int textHaloYOffset = 1;                                                                                                                                                                                                                                                  
         ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:750:9: warning: unused variable ‘textHaloSize’ [-Wunused-variable]
     int textHaloSize = 2;                                                                                                                                                                                                                                                     
         ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp: At global scope:
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:893:62: warning: unused parameter ‘repaintArea’ [-Wunused-parameter]
 void DecorationButton::paint(QPainter *painter, const QRect &repaintArea)                                                                                                                                                                                                     
                                                              ^
/home/richard/Downloads/smaragd-0.1.0/src/kwin_smaragd.cpp:935:66: warning: unused parameter ‘repaintArea’ [-Wunused-parameter]
 void DecorationButton::paintGlow(QPainter *painter, const QRect &repaintArea)                                                                                                                                                                                                 
                                                                  ^
[ 26%] Building C object src/CMakeFiles/kwin_smaragd.dir/emerald.c.o                                                                                                                                                                                                           
/home/richard/Downloads/smaragd-0.1.0/src/emerald.c: In function ‘load_font_setting’:
/home/richard/Downloads/smaragd-0.1.0/src/emerald.c:126:62: warning: unused parameter ‘fd’ [-Wunused-parameter]
 void load_font_setting(GKeyFile * f, PangoFontDescription ** fd, const gchar * key, const gchar * sect)                                                                                                                                                                       
                                                              ^
/home/richard/Downloads/smaragd-0.1.0/src/emerald.c: In function ‘rounded_rectangle’:
/home/richard/Downloads/smaragd-0.1.0/src/emerald.c:156:27: warning: unused parameter ‘ws’ [-Wunused-parameter]
         window_settings * ws,                                                                                                                                                                                                                                                 
                           ^
[ 33%] Building C object src/CMakeFiles/kwin_smaragd.dir/main.c.o
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_help_button’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:271:40: warning: unused parameter ‘d’ [-Wunused-parameter]
 static void draw_help_button(decor_t * d, cairo_t * cr, double s)
                                        ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:271:64: warning: unused parameter ‘s’ [-Wunused-parameter]
 static void draw_help_button(decor_t * d, cairo_t * cr, double s)
                                                                ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_close_button’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:290:41: warning: unused parameter ‘d’ [-Wunused-parameter]
 static void draw_close_button(decor_t * d, cairo_t * cr, double s)
                                         ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_max_button’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:311:39: warning: unused parameter ‘d’ [-Wunused-parameter]
 static void draw_max_button(decor_t * d, cairo_t * cr, double s)
                                       ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_unmax_button’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:330:41: warning: unused parameter ‘d’ [-Wunused-parameter]
 static void draw_unmax_button(decor_t * d, cairo_t * cr, double s)
                                         ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_min_button’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:351:39: warning: unused parameter ‘d’ [-Wunused-parameter]
 static void draw_min_button(decor_t * d, cairo_t * cr, double s)
                                       ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_pixbuf’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:421:25: warning: unused parameter ‘pixbuf’ [-Wunused-parameter]
 draw_pixbuf(GdkPixbuf * pixbuf, cairo_t * cr,
                         ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:421:43: warning: unused parameter ‘cr’ [-Wunused-parameter]
 draw_pixbuf(GdkPixbuf * pixbuf, cairo_t * cr,
                                           ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:422:21: warning: unused parameter ‘x’ [-Wunused-parameter]
             gdouble x, gdouble y, gdouble x2, gdouble y2, gdouble alpha)
                     ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:422:32: warning: unused parameter ‘y’ [-Wunused-parameter]
             gdouble x, gdouble y, gdouble x2, gdouble y2, gdouble alpha)
                                ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:422:43: warning: unused parameter ‘x2’ [-Wunused-parameter]
             gdouble x, gdouble y, gdouble x2, gdouble y2, gdouble alpha)
                                           ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:422:55: warning: unused parameter ‘y2’ [-Wunused-parameter]
             gdouble x, gdouble y, gdouble x2, gdouble y2, gdouble alpha)
                                                       ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:422:67: warning: unused parameter ‘alpha’ [-Wunused-parameter]
             gdouble x, gdouble y, gdouble x2, gdouble y2, gdouble alpha)
                                                                   ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: In function ‘draw_button_with_glow_alpha_bstate’:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:435:41: warning: unused parameter ‘y1’ [-Wunused-parameter]
                                    gint y1, gdouble button_alpha,
                                         ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c: At top level:
/home/richard/Downloads/smaragd-0.1.0/src/main.c:55:13: warning: ‘get_b_t_offset’ defined but not used [-Wunused-function]
 static gint get_b_t_offset(gint b_t)
             ^
/home/richard/Downloads/smaragd-0.1.0/src/main.c:408:12: warning: ‘get_b_state’ defined but not used [-Wunused-function]
 static int get_b_state(decor_t * d, int button)
            ^
[ 40%] Building CXX object src/CMakeFiles/kwin_smaragd.dir/shadowengine.cpp.o
[ 46%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/legacy.c.o
[ 53%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/line.c.o
[ 60%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/oxygen.c.o
[ 66%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/pixmap.c.o
[ 73%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/truglass.c.o
[ 80%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/vrunner.c.o
[ 86%] Building C object src/CMakeFiles/kwin_smaragd.dir/engines/zootreeves.c.o
/home/richard/Downloads/smaragd-0.1.0/src/engines/zootreeves.c: In function ‘rounded_rectangle_independent’:
/home/richard/Downloads/smaragd-0.1.0/src/engines/zootreeves.c:118:27: warning: unused parameter ‘ws’ [-Wunused-parameter]
         window_settings * ws,
                           ^
/home/richard/Downloads/smaragd-0.1.0/src/engines/zootreeves.c: In function ‘rounded_square’:
/home/richard/Downloads/smaragd-0.1.0/src/engines/zootreeves.c:316:27: warning: unused parameter ‘ws’ [-Wunused-parameter]
         window_settings * ws,
                           ^
[ 93%] Building CXX object src/CMakeFiles/kwin_smaragd.dir/kwin_smaragd_automoc.cpp.o
[100%] Linking CXX shared module kwin_smaragd.so
[100%] Built target kwin_smaragd
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0/build$ sudo make install

**Comment by cinnamoner** [comment]
@cfeck 

[  6%] Automatic moc for target kwin_smaragd
[  6%] Built target kwin_smaragd_automoc
[100%] Built target kwin_smaragd
Install the project...
-- Install configuration: ""
-- Installing: /usr/lib/x86_64-linux-gnu/qt5/plugins/org.kde.kdecoration2/kwin_smaragd.so
-- Installing: /etc/xdg/smaragd.knsrc

**Comment by cinnamoner** [comment]
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0$ cd smaragd
bash: cd: smaragd: No such file or directory
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0$ mkdir build
mkdir: cannot create directory ‘build’: File exists
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0$ cd build
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0/build$ cmake -DCMAKE_INSTALL_PREFIX=`kde4-config --prefix` ..
-- Configuring done
-- Generating done
-- Build files have been written to: /home/richard/Downloads/smaragd-0.1.0/build
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0/build$ make
[  6%] Automatic moc for target kwin_smaragd
[  6%] Built target kwin_smaragd_automoc
[100%] Built target kwin_smaragd
richard@richard-Veriton-L670G:~/Downloads/smaragd-0.1.0/build$ sudo make install

**Comment by cinnamoner** [comment]
@cfeck What else?

**Comment by cinnamoner** [comment]
-DCMAKE_INSTALL_PREFIX=kde4-config --prefix ..

@cfeck So, How do I get rid of that?

**Comment by cinnamoner** [comment]
There's gotta be a simpler way.

**Comment by cfeck** [comment]
You already installed it.

Now logout, and when you login again, you can use systemsettings5 to change the window decoration to Smaragd. You should have an emerald theme installed in .emerald/theme, but if you already had used Emerald in the same home directory, there will already be the last selected theme.

**Comment by cinnamoner** [comment]
@cfeck @BigDaddyonTech @MichaelTunnell

Proof of concept achieved. The thing works. Some minor glitches here n there (cause: yet unknown).
![screenshot_20170308_103858](https://cloud.githubusercontent.com/assets/15638605/23696477/7f93df48-03ec-11e7-8225-0e0c7266dccb.png)
![screenshot_20170308_103910](https://cloud.githubusercontent.com/assets/15638605/23696490/90f623fe-03ec-11e7-851c-7b3d85b95307.png)

@cfeck Thank you very much. You efforts are much appreciated. You should definitely perfect it even further.

**Comment by cinnamoner** [comment]
More detailed feedback coming soon ...

**Comment by cinnamoner** [comment]
**Disclaimer**: Should anyone want to replicate it, he/she does it on his/her risk. At least make sure you use a test machine and definitely not your daily driver.

**Comment by cinnamoner** [comment]
**Disclaimer:** Should anyone want to replicate it, he/she does it on his/her risk. At least make sure you use a test machine and definitely not your daily driver. Seems smooth, stable and snappier than Aurorae, but I need further analysis.

**Comment by cinnamoner** [comment]
@cfeck I'll invest some time into this project. It's definitely worth it. Thanks. We'll keep in touch 👍

**Comment by cinnamoner** [comment]
https://kdepepo.wordpress.com/2016/12/23/smaragd-for-plasma-5/

*Although somewhat intimidating at first glance (or second, or third) for noobs like myself, the above link and associated links are definitely worth a thorough read, should one want to get a deeper understanding of what this wannabe tutorial has dealt with.*

**Comment by cinnamoner** [comment]
*Feel free to ask!*

**Comment by cinnamoner** [comment]
So, in order to build Smaragd on KDE Neon, one needs to: 

*sudo apt-get install git build-essential cmake qtbase5-dev libkf5config-dev libkf5coreaddons-dev libkf5i18n-dev libkf5widgetsaddons-dev libkdecorations2-dev libglib2.0-dev libcairo2-dev extra-cmake-modules gettext*

Now, I'll check those packages one-by-one. I need to know more about them.

**Comment by cinnamoner** [comment]
Note: *Used themes are illustrative in scope. By no means should one conclude I'm interested in Smaragd for the sole purpose of running Windows-inspired themes. Nope.*

**Comment by cinnamoner** [comment]
@cfeck I've posted some questions on the KDE forum. Just want you to know about it, that's all.

**Comment by cinnamoner** [comment]
@cfeck I've started from scratch again and installed required packages using:

*sudo apt-get install git build-essential cmake qtbase5-dev libkf5config-dev libkf5coreaddons-dev libkf5i18n-dev libkf5widgetsaddons-dev libkdecorations2-dev libglib2.0-dev libcairo2-dev extra-cmake-modules gettext*

Everything installed smoothly.

Now, I'm preparing to install Smaragd again, but you had said:

> Additionally, do not use kde4-config, that is invalid in Plasma5. Just use /usr as the prefix.

The install instructions read:

> cd smaragd
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=`kde4-config --prefix` ..
make
sudo make install

So, which is the correct line for this in my case?
cmake -DCMAKE_INSTALL_PREFIX=`kde4-config --prefix` ..

Just post the changed line, please. Thank you. 'Cause I want it to be accurate.

**Comment by cinnamoner** [comment]
cd smaragd
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=`kde4-config --prefix` ..
make
sudo make install

**Comment by cinnamoner** [comment]
So, one doesn't need to download any packages. One only needs the terminal.

**Comment by cinnamoner** [comment]
![success](https://cloud.githubusercontent.com/assets/15638605/23822525/f954832a-0656-11e7-962f-edf21790d211.png)

**Comment by cinnamoner** [comment]
@cfeck What are those warnings?

![warning_01](https://cloud.githubusercontent.com/assets/15638605/23822654/d50afb90-0659-11e7-9565-b985d5846001.png)
![warning_02](https://cloud.githubusercontent.com/assets/15638605/23822658/e114394c-0659-11e7-9917-5aa63bb26454.png)

**Comment by cinnamoner** [comment]
@BigDaddyonTech The thing work. I'll be trying out some themes from the KDE Store. Here's just one.

![works](https://cloud.githubusercontent.com/assets/15638605/23822806/f97fbb02-065c-11e7-975f-1f4aa8693863.png)

**Comment by cinnamoner** [comment]
> In the future, GHNS support will be added to ease installation.

That would be great.

**Comment by cinnamoner** [comment]
When this tutorial will be refined enough, I'm planning but not promising to start one on Dekorator.

**Comment by cinnamoner** [comment]
@BigDaddyonTech Here's another.

![vox01](https://cloud.githubusercontent.com/assets/15638605/23824133/ae070dca-0679-11e7-8814-f2380e4896aa.png)
![vox02](https://cloud.githubusercontent.com/assets/15638605/23824136/b71c80c0-0679-11e7-9e32-f9fb8b89872b.png)

**Comment by BigDaddyonTech** [comment]
Is this something that can easily be switched on and off? Are you experiencing any issues by using it?

**Comment by cinnamoner** [comment]
> Is this something that can easily be switched on and off? 

One can easily switch from an Emerald theme to any Aurorae, but one can only have 1 Emerald in the "Windows Decorations" at one time.

> Are you experiencing any issues by using it?

So far so good. No issues.

**Comment by cinnamoner** [comment]
@BigDaddyonTech Switching works seamlessly ...

![wd2](https://cloud.githubusercontent.com/assets/15638605/23824459/f6ac4de6-067f-11e7-9d55-3b02b12bf6bf.png)

**Comment by cinnamoner** [comment]
> one can only have 1 Emerald in the "Windows Decorations" at one time

@cfeck Am I Correct?

Also, I think Breeze is neither Emerald or Aurorae.

**Comment by cinnamoner** [comment]
@BigDaddyonTech Screenshot update right above. See what I mean? Not a deal breaker though.

**Comment by cinnamoner** [comment]
Posted themes were meant for illustrative purposes. Might or might not reflect my good taste :)

**Comment by cinnamoner** [comment]
I've tried more than 70 Emerald themes. Smaragd runs stable and feels sappier than Aurorae.

**Comment by BigDaddyonTech** [comment]
My dream is to get @Elbullazul Windows 10 theme working in KDE

**Comment by cinnamoner** [comment]
> My dream is to get @Elbullazul Windows 10 theme working in KDE

Sounds interesting.

**Comment by cinnamoner** [comment]
Might make a tutorial on ... Dekorator? I that the name of the thing? But I'm quite satisfied with Smaragd. Enthused even.

**Comment by cinnamoner** [comment]
Quote from cfeck:

> If you are using transparent themes, I suggest to enable the Blur desktop effect. Needs a working OpenGL driver, though.

@cfeck I'll see what I can do. Further support might be needed. Thanks. Have I said you're awesome? Guess I have.

**Comment by cinnamoner** [comment]
@MichaelTunnell Not watching that thread? Understandable. How about this one then?

**Comment by MichaelTunnell** [comment]
> @MichaelTunnell Not watching that thread? Understandable. How about this one then?

I do not watch threads I'm not participating in. 

If you want to ask me something specific, feel free to do so in the message to me.

**Comment by cinnamoner** [comment]
> If you want to ask me something specific, feel free to do so in the message to me.

@MichaelTunnell Will do. Thanks.

**Comment by cinnamoner** [comment]
So, in order to achieve the blur effect in KDE Neon, one goes:
_System Settings > Workspace > Desktop Behavior > Desktop Effects > Appearance > Blur_

![screenshot_20170316_004154](https://cloud.githubusercontent.com/assets/15638605/23974005/98ad75d4-09e1-11e7-8356-f66e1ebee942.png)

One can adjust the _strength_ of the blur effect to his/her desire

![screenshot_20170316_004102](https://cloud.githubusercontent.com/assets/15638605/23974164/6e8d7442-09e2-11e7-9ff6-cebac98d4963.png)

And, one can get something similar to that

![screenshot_20170315_211442](https://cloud.githubusercontent.com/assets/15638605/23974193/9f38f756-09e2-11e7-953a-491725699455.png)

So, in a nutshell, that's it. Enjoy.

**Comment by cinnamoner** [comment]
Further questions? Fell free to ask.

**Comment by cinnamoner** [comment]
@cfeck @BigDaddyonTech @feren Now that I've played with it a lot, I find Smaragd significantly snappier/faster than Aurorae.

**Comment by cinnamoner** [comment]
`git clone git://anongit.kde.org/smaragd.git cd smaragd git checkout kdecoration2 mkdir build && cd build cmake -DCMAKE_INSTALL_PREFIX=/usr .. make && sudo make install`

@cfeck Is that the correct way? If I coppypaste that into the terminal, will it work? I'm not sure about the spaces. Is it correctly spaced like that?

**Comment by cfeck** [comment]
No, it is not correct. These are multiple commands, and you have to start them line-by-line.

**Comment by cinnamoner** [comment]
@cfeck Okay. Thanks. I'll try again.

`git clone git://anongit.kde.org/smaragd.git`
`cd smaragd`
`git checkout kdecoration2`
`mkdir build && cd build`
`cmake -DCMAKE_INSTALL_PREFIX=/usr ..`
`make && sudo make install`

Is it correct now?

**Comment by cfeck** [comment]
Yes.

**Comment by cinnamoner** [comment]
No answer usually means yes.

**Comment by cinnamoner** [comment]
`sudo apt-get install git build-essential cmake qtbase5-dev libkf5config-dev libkf5coreaddons-dev libkf5i18n-dev libkf5widgetsaddons-dev libkdecorations2-dev libglib2.0-dev libcairo2-dev extra-cmake-modules gettext`

@cfeck And this one?

**Comment by cfeck** [comment]
This is a single command, so it needs to be a single line.

Why are you going through all this again?

**Comment by cinnamoner** [comment]
@cfeck Because I'm a perfectionist and I want to refine this tutorial. I'm also learning a thing or two in the process. Last but not least, I love Smaragd. It's fantastic work. I couldn't believe what it's capable of. When I had seen it working I was ecstatic. It's almost unbelievable. And it's snappier than Aurorae. I appreciate your help tremendously. Thank you.

**Comment by cinnamoner** [comment]
@BigDaddyonTech This is way more exciting than the return of the bouncy ball. 😄

**Comment by cinnamoner** [comment]
I don't think this tutorial needs improvement. It's the flaws that make it perfect. It happened in real time and that adds value to it. I'm not rewriting history.

**Comment by cinnamoner** [comment]
@cfeck https://github.com/KDE/extra-cmake-modules/commit/97ba63aafa510e64c78a529cb5e88eebd2e97967
I hope Smaragd won't be affected.
